### PR TITLE
Str trim void

### PR DESCRIPTION
**Pull request by radare**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](../DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](../DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](../../radare2book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: libr/util/str_trim.c [hunk 1]
```diff
@@ -82,6 +82,10 @@ R_API char *r_str_trim_dup(const char *str) {
 
 R_API void r_str_trim(char *str) {
 	r_return_if_fail (str);
+#if 1
+	r_str_trim_head (str);
```
Comment (ret2libc) [inline libr/util/str_trim.c]: why calling this if in the next line you call r_str_trim_head_tail, which does both head and tail?

File: libr/util/str_trim.c [hunk 2]
```diff
@@ -82,6 +82,10 @@ R_API char *r_str_trim_dup(const char *str) {
 
 R_API void r_str_trim(char *str) {
```
Comment (ret2libc) [inline libr/util/str_trim.c]: this seems just a dup of `r_str_trim_head_tail`

File: libr/util/str_trim.c [hunk 3]
```diff
@@ -101,81 +105,62 @@ R_API void r_str_trim(char *str) {
...
-	return r_str_trim_tail (r_str_trim_head (str));
+R_API void r_str_trim_head_tail(char *str) {
+	r_str_trim_head (str);
+	r_str_trim_tail (str);
 }
 
-// no copy, like trim_head+tail but with trim_head_ro
+// no copy, like trim_head+tail but with trim_head_ro, beware heap issues
+// TODO: rename to r_str_trim_weak() ?
 R_API char *r_str_trim_nc(char *str) {
```
Comment (ret2libc) [inline libr/util/str_trim.c]: what does `nc` stand for?

**Comment by radare** [comment]
> On 24 Feb 2020, at 09:45, Riccardo Schirone <notifications@github.com> wrote:
> 
> ﻿
> @ret2libc requested changes on this pull request.
> 
> I'm ok with the changes in r_str.h and str.c, mostly, but the PR needs to be greenify, you should be careful where you put r_return_, because in some cases NULL values are actually passed. So, I'd say to keep the PR small and focused on improving just r_str_trim_, don't add other stuff but just do those small changes.
> 

Im trying to find which of the changes is making this pr red
> 
> In libr/util/str_trim.c:
> 
> > @@ -82,6 +82,10 @@ R_API char *r_str_trim_dup(const char *str) {
>  
>  R_API void r_str_trim(char *str) {
>  	r_return_if_fail (str);
> +#if 1
> +	r_str_trim_head (str);
> why calling this if in the next line you call r_str_trim_head_tail, which does both head and tail?
> 


Because its dupe 
> 
> In libr/util/str_trim.c:
> 
> > @@ -82,6 +82,10 @@ R_API char *r_str_trim_dup(const char *str) {
>  
>  R_API void r_str_trim(char *str) {
> this seems just a dup of r_str_trim_head_tail
> 
> In libr/util/str_trim.c:
> 
> >  }
>  
> -// no copy, like trim_head+tail but with trim_head_ro
> +// no copy, like trim_head+tail but with trim_head_ro, beware heap issues
> +// TODO: rename to r_str_trim_weak() ?
>  R_API char *r_str_trim_nc(char *str) {
> what does nc stand for?
> 

No-copy

File: libr/util/str_trim.c (hunk 1 above)
Comment (radare) [reply libr/util/str_trim.c]: because thats what r_str_trim() do. ts an alias for head_tail

Comment (ret2libc) [reply libr/util/str_trim.c]: If it is an alias, just make it a define in the r_str.h header  or implement it as:
```C
R_API void r_str_trim(char *str) {
   r_str_trim_head_tail (str);
}
```
don't duplicate the implementation in both functions.

Comment (radare) [reply libr/util/str_trim.c]: nvm i just removed the dupe

File: libr/util/str_trim.c (hunk 2 above)
Comment (radare) [reply libr/util/str_trim.c]: yep indeed. this is fixed now

File: libr/util/str_trim.c (hunk 3 above)
Comment (radare) [reply libr/util/str_trim.c]: no-copy, its a shitty name but i didnt decided for it, and i think its worth having it for performance reasons

**Comment by radare** [comment]
r_str_trim_head_ro -> needs to be renamed to r_str_skip_spaces() or find a better name, but i would be happy if i could just find the reason for the redness and merge before dong more changes like this, but the api is cleaner now

**Comment by radare** [comment]
if any assert returns null i would get a warning in travis or in the shell

**Comment by radare** [comment]
Defines are stupid. This is a refactoring not a hack

**Comment by ret2libc** [comment]
There are still these errors: https://travis-ci.com/radareorg/radare2/jobs/290752798#L703

**Comment by radare** [comment]
@ret2libc i cant reproduce them in local

File: libr/core/cconfig.c [hunk 4]
```diff
@@ -3261,6 +3261,9 @@ R_API int r_core_config_init(RCore *core) {
 	SETCB ("dir.home", p, &cb_dirhome, "Path for the home directory");
 	free (p);
 	p = r_sys_getenv (R_SYS_TMP);
+	if (!p) {
```
Comment (ret2libc) [inline libr/core/cconfig.c]: just do as done in other places:
```C
SETCB ("dir.tmp", p? p: "", &cb_dirtmp, "Path of the tmp directory");
```

Comment (radare) [reply libr/core/cconfig.c]: yeah thats simpler, lets see if travis is happy and ill commit this

**Pull request merged**